### PR TITLE
Allow use unitless in mixin lib-line-height

### LINKED ISSUE #25592: @font-size-unit-convert affects unitless values
### Preconditions (*)

Magento v2.3.1 & 2.3-develop

### Steps to reproduce (*)
Compile this LESS under either blank or Luma:

```less
.foo-bar {
  .lib-line-height(1);
}
```

### Expected result (*)
```less
.foo-bar {
   line-height: 1;
}
```

### Actual result (*)
```less
.foo-bar {
   line-height: 0.1rem;
}
```

This is the underlying mixin powering many, many lib- mixins across the site, for instance `lib-button()`, and thus this issue pertains to e.g `@button__line-height: 1;`.

As you follow the issue, you realize this is systemic in the way the plumbing for these mixins operates:

https://github.com/magento/magento2/blob/54742d5150fa97d08defd89226ddf5cc87aefa85/lib/web/css/source/lib/_utilities.less#L14-L23

`@font-size-unit` is defaulted to `rem`, and thus the mixin will _always_ convert to this, even if it's not needed. I'll note that this behavior is disabled when `@font-size-unit-convert: false`, though this is not the default under either blank or Luma.

A simple fix would be to add a guard clause utilizing LESS' built-in [`get-unit()`](http://lesscss.org/functions/#misc-functions-get-unit) function, which returns falsey if the value passed is unitless.

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

Before this PR magento not properly set line-height with unitless value

.lib-line-height(1) => output line-height: 0.1rem;
Expected result => output line-height: 1;

After patch we can use like below
.lib-line-height(normal);
.lib-line-height(1.525);
.lib-line-height(15%);
.lib-line-height(2);
.lib-line-height(24px);
.lib-line-height(1.5rem);
.lib-line-height(2.5em);

### Fixed Issues (if relevant)

1. Fixes magento/magento2#25592

### Manual testing scenarios (*)

1. Go to theme less file. Example app/design/frontend/Magento/blank/Magento_Theme/web/css/source/_module.less
2. Add style
body .page-title {.lib-line-height(1.5);}
3. Run setup static and clear cache
4. Inspect page title element in page by browser to view result styles

### Questions or comments

CC: @Leland Is this PR satisfied you or help ?
### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run WebAPI Tests, Unit Tests

File: lib/web/css/source/lib/_typography.less [hunk 1]
```diff
@@ -63,8 +63,16 @@
...
+//  _____________________________________________
+
+.lib-line-height(@heightValue) when not (@heightValue = false) and not (@heightValue = normal) and not (ispercentage(@heightValue)) and (isnumber(@heightValue)) {
+    .lib-css(line-height, @heightValue);
+}
+
+.lib-line-height(@heightValue) when not (@heightValue = false) and not (@heightValue = normal) and not (ispercentage(@heightValue)) and not (isnumber(@heightValue)) {
     .lib-font-size-value(@heightValue);
     .lib-css(line-height, @fontValue);
 }
```
Comment (Leland) [inline lib/web/css/source/lib/_typography.less]: My local environments need a tune up before I can test this, but this looks promising! Can you make sure to add some comments explaining what all these booleans are doing? No future dev will be able to make heads or tails of these, speaking from experience looking at M2 LESS 😆

Something just along the lines of like

```less
// Check for unitless values (e.g. 0, 1, etc)

**Review by Leland — CHANGES_REQUESTED** [review]
.

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B, Functional Tests CE

**Review by Leland — APPROVED** [review]
LGTM 🤘

**Comment by mrtuvn** [comment]
rebased code branch by mainline branch

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Unit Tests, Functional Tests EE, Functional Tests CE, Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE, WebAPI Tests

**Comment by mrtuvn** [comment]
Rebased code

**Comment by mrtuvn** [comment]
rebased code
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B, Functional Tests CE

**Comment by engcom-Bravo** [comment]
Hi @mrtuvn,

We are closing this PR.The PR is in Draft since longtime.

Once you are ready Please feel free to reopen it to process the PR further.

Thanks

**Pull request closed without merging**